src/core/stdc/stdio.d

```diff
@@ -1410,16 +1446,85 @@ else version (CRuntime_Microsoft)
     int  vsnprintf(scope char* s, size_t n, scope const char* format, va_list arg);
   }
 
+    //
+    // Microsoft under-the-hood C I/O functions. Uses _iobuf* for the unshared
+    // version of FILE*, usable when the FILE is locked.
+    //
+    import core.stdc.stddef : wchar_t;
+    import core.stdc.wchar_ : wint_t;
+
+    version (MinGW)
+    {
+        private int _filbuf(_iobuf*);
+        private int _flsbuf(int, _iobuf*);
+
+        extern (D)
+        {
+            ///
+            int _fputc_nolock()(int c, _iobuf* fp)
+            {
+                pragma(inline, true);
+                fp._cnt--;
+                if (fp._cnt >= 0)
+                {
+                    *fp._ptr++ = cast(char)c;
+                    return c & 0xFF;
+                }
+                else
+                    return _flsbuf(c, fp);
+            }
+            ///
+            int _fgetc_nolock()(_iobuf* fp)
+            {
+                pragma(inline, true);
+                fp._cnt--;
+                if (fp._cnt >= 0)
+                {
+                    immutable c = *fp._ptr++;
+                    return c & 0xFF;
+                }
+                else
+                    return _filbuf(fp);
+            }
+            ///
+            wint_t _fputwc_nolock()(wchar_t c, _iobuf* fp)
+            {
+                pragma(inline, true);
+                import core.stdc.wchar_ : fputwc;
+                return fputwc(c, cast(shared) fp);
+            }
+            ///
+            wint_t _fgetwc_nolock()(_iobuf* fp)
+            {
+                pragma(inline, true);
+                import core.stdc.wchar_ : fgetwc;
+                return fgetwc(cast(shared) fp);
+            }
```

Comment:
I'm not bothered if instead we move this to be instead handled in phobos.

Reply:
I convinced myself to remove it.  Will put these helper functions in Phobos std.stdio.